yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -353,7 +353,7 @@ private[yarn] class YarnAllocator(
 
     } else if (missing < 0) {
       val numToCancel = math.min(numPendingAllocate, -missing)
-      logInfo(s"Canceling requests for $numToCancel executor containers")
+      logInfo(s"Canceled requests for $numToCancel executor container(s)")
```

Comment:
But, it is the fact that it is now 614 and not 619 that causes 5 to be cancelled right? If it first says 5 are being cancelled, then says why, is that not out of order?